RecoMuon/MeasurementDet/src/MuonDetLayerMeasurements.cc

```diff
@@ -150,34 +150,51 @@ MuonRecHitContainer MuonDetLayerMeasurements::recHits(const GeomDet* geomDet,
 
 	// Create the chamber Id
 	GEMDetId chamberId(geoId.rawId());
-
-	if (chamberId.layer() == 0){
-	  // Get the GEMSegment which relies on this chamber
-	  GEMSegmentCollection::range range = theGEMSegments->get(chamberId);
-	  LogDebug("Muon|RecoMuon|MuonDetLayerMeasurements") << "Number of GEM rechits available =  " << theGEMSegments->size()
-							     <<", from chamber: "<< chamberId<<std::endl;
-	  // Create the MuonTransientTrackingRecHit
-	  for (GEMSegmentCollection::const_iterator rechit = range.first;
-	       rechit!=range.second; ++rechit){
-	    LogDebug("Muon|RecoMuon|MuonDetLayerMeasurements") << "On GEM iteration " <<std::endl;
-	    result.push_back(MuonTransientTrackingRecHit::specificBuild(geomDet,&*rechit));
-	  }
+	if (chamberId.layer() != 0)
+	  return result;
```

Comment:
STA track finder also uses MuonDetLayerMeasurements, but since the detector layers have been changed to 0 (superChamber) only, this case wouldnt happen.